### PR TITLE
Update 2.0 docs to use docker image for unstable releases

### LINKED ISSUE #2283: build: Publish a separate docker image for unstable releases
PR: https://github.com/cockroachdb/cockroach/pull/20331

### PR DESCRIPTION
**Pull request by jseldess**
Fixes #2283

### CODE REVIEW

File: _includes/start_in_docker/v2.0/mac-linux-steps.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+## Before You Begin
```
Comment (a-robinson) [inline _includes/start_in_docker/v2.0/mac-linux-steps.md]: It seems silly to duplicate this entire file based on the version that's being run, but I take it there wasn't a good way to share it?

File: _includes/start_in_docker/v2.0/mac-linux-steps.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+Make sure you have already [installed the official CockroachDB Docker image](install-cockroachdb.html).
```
Comment (a-robinson) [inline _includes/start_in_docker/v2.0/mac-linux-steps.md]: Should this say "the unstable CockroachDB Docker image"?

File: _includes/start_in_docker/v2.0/mac-linux-steps.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+## Step 2. Start the first node
+
+~~~ shell
+$ docker run -d \
+--name=roach1 \
+--hostname=roach1 \
+--net=roachnet \
+-p 26257:26257 -p 8080:8080  \
+-v "${PWD}/cockroach-data/roach1:/cockroach/cockroach-data"  \
+cockroachdb/cockroach-unstable start --insecure
```
Comment (a-robinson) [inline _includes/start_in_docker/v2.0/mac-linux-steps.md]: Is there a reason we aren't including the version tag here, a la `cockroachdb/cockroach-unstable:{{page.release_info.version}}`?

This comment applies to all of the examples.

File: _includes/start_in_docker/v2.0/mac-linux-steps.md (hunk 1 above)
Comment (jseldess) [reply _includes/start_in_docker/v2.0/mac-linux-steps.md]: You're right. I put in place a global variable. PTAL.

File: _includes/start_in_docker/v2.0/mac-linux-steps.md (hunk 2 above)
Comment (jseldess) [reply _includes/start_in_docker/v2.0/mac-linux-steps.md]: Is it ok to accept this slightly misleading text for the sake of not doing something unnecessarily complicated with variables?

File: _includes/start_in_docker/v2.0/mac-linux-steps.md (hunk 3 above)
Comment (jseldess) [reply _includes/start_in_docker/v2.0/mac-linux-steps.md]: Done.

File: _includes/start_in_docker/v2.0/mac-linux-steps.md (hunk 2 above)
Comment (a-robinson) [reply _includes/start_in_docker/v2.0/mac-linux-steps.md]: Seems ok to me.

**Pull request merged**